xbmc/platform/android/activity/EventLoop.cpp

```diff
@@ -23,33 +23,30 @@ CEventLoop::CEventLoop(android_app* application)
   m_application->onInputEvent = inputCallback;
 }
 
-void CEventLoop::run(IActivityHandler &activityHandler, IInputHandler &inputHandler)
+void CEventLoop::run(IActivityHandler& activityHandler, IInputHandler& inputHandler)
 {
-  int ident;
-  int events;
-  struct android_poll_source* source;
-
   m_activityHandler = &activityHandler;
   m_inputHandler = &inputHandler;
 
   CXBMCApp::android_printf("CEventLoop: starting event loop");
-  while (true)
+
+  while (!m_application->destroyRequested)
   {
-    // We will block forever waiting for events.
-    while ((ident = ALooper_pollAll(-1, NULL, &events, (void**)&source)) >= 0)
+    android_poll_source* source = nullptr;
+    auto result = ALooper_pollOnce(-1, nullptr, nullptr, reinterpret_cast<void**>(&source));
+
+    if (result == ALOOPER_POLL_ERROR)
     {
-      // Process this event.
-      if (source != NULL)
-        source->process(m_application, source);
-
-      // Check if we are exiting.
-      if (m_application->destroyRequested)
-      {
-        CXBMCApp::android_printf("CEventLoop: we are being destroyed");
-        return;
-      }
+      CXBMCApp::android_printf("CEventLoop: ALooper_pollOnce returned an error");
+      break;
     }
+
+    // Process this event.
+    if (source != nullptr)
+      source->process(m_application, source);
   }
+
+  CXBMCApp::android_printf("CEventLoop: we are being destroyed");
```

Comment:
Clog?

Reply:
If events occur before CLog initialisation, a crash occurs.

`CXBMCApp::android_printf()` takes this into account and uses CLog if is initialised.